save-api/src/main/kotlin/org/cqfn/save/api/AutomaticTestInitializator.kt

```diff
@@ -0,0 +1,265 @@
+    }
+
+    /**
+     * Build execution request for standard mode according provided configuration
+     *
+     * @param requestUtils
+     * @param userProvidedTestSuites test suites, specified by user in config file
+     */
+    private suspend fun buildExecutionRequestForStandardSuites(
+        requestUtils: RequestUtils,
+        userProvidedTestSuites: List<String>
+    ): Pair<Organization, ExecutionRequestForStandardSuites> {
+        val (organization, project) = getOrganizationAndProject(requestUtils)
+
+        return organization to ExecutionRequestForStandardSuites(
+            project = project,
+            testsSuites = userProvidedTestSuites,
+            sdk = Jdk("11"),
+            execCmd = evaluatedToolProperties.execCmd,
+            batchSizeForAnalyzer = evaluatedToolProperties.batchSize
+        )
+    }
+
+    /**
+     * Verify for correctness test suites, specified by user, return them or nothing if they are incorrect
+     *
+     * @param requestUtils
+     * @return list of test suites or nothing
+     */
+    private suspend fun verifyTestSuites(requestUtils: RequestUtils): List<String>? {
+        val userProvidedTestSuites = evaluatedToolProperties.testSuites.split(";")
+        if (userProvidedTestSuites.isEmpty()) {
+            log.error("Set of test suites couldn't be empty in standard mode!")
+            return null
+        }
+
+        val existingTestSuites = requestUtils.getStandardTestSuites().map { it.name }
+
+        userProvidedTestSuites.forEach {
+            if (it !in existingTestSuites) {
+                log.error("Incorrect standard test suite $it, available are $existingTestSuites")
+                return null
+            }
+        }
+        return userProvidedTestSuites
+    }
+
+    /**
+     * Return pair of organization and project according information from config file
+     *
+     * @param requestUtils
+     */
+    @Suppress("UnsafeCallOnNullableType")
+    private suspend fun getOrganizationAndProject(requestUtils: RequestUtils): Pair<Organization, Project> {
+        val organization = requestUtils.getOrganizationByName(evaluatedToolProperties.organizationName)
+        val project = requestUtils.getProjectByNameAndOrganizationId(evaluatedToolProperties.projectName, organization.id!!)
+        return organization to project
+    }
+
+    /**
+     * Get results for current [executionRequest] and [organizationId]:
+     * sending requests, which checks current state of execution, until it will be finished, or timeout will be reached
+     *
+     * @param requestUtils
+     * @param executionRequest
+     * @param organizationId
+     */
+    @Suppress("MagicNumber")
+    private suspend fun getExecutionResults(
+        requestUtils: RequestUtils,
+        executionRequest: ExecutionRequestBase,
+        organizationId: Long
+    ): ExecutionDto? {
+        // Execution should be processed in db after submission, so wait little time
+        Thread.sleep(1_000)
```

Comment:
You already built the client on coroutines, so using `delay` is much better

Reply:
Thanks. fixed